Examples/Scripts/MaterialMapping/materialComposition.C

```diff
@@ -108,128 +108,131 @@ void materialComposition(const std::string& inFile, const std::string& treeName,
   // Open the input file & get the tree
   auto inputFile = TFile::Open(inFile.c_str());
   auto inputTree = dynamic_cast<TTree*>(inputFile->Get(treeName.c_str()));
-  if (inputTree != nullptr) {
-    // Get the different atomic numbers
-    TCanvas* materialCanvas =
-        new TCanvas("materialCanvas", "Materials", 100, 100, 620, 400);
-    materialCanvas->cd();
-    // Draw all the atomic elements & get the histogram
-    inputTree->Draw("mat_A>>hA(100,0.5,100.5)");
-    TH1F* histA = dynamic_cast<TH1F*>(gDirectory->Get("hA"));
-    histA->Draw();
-
-    auto outputFile = TFile::Open(outFile.c_str(), "recreate");
-
-    float v_eta = 0;
-    float v_phi = 0;
-    std::vector<float>* stepLength = new std::vector<float>;
-    std::vector<float>* stepX0 = new std::vector<float>;
-    std::vector<float>* stepL0 = new std::vector<float>;
-    std::vector<float>* stepA = new std::vector<float>;
-
-    std::vector<float>* startX = new std::vector<float>;
-    std::vector<float>* startY = new std::vector<float>;
-    std::vector<float>* startZ = new std::vector<float>;
-
-    std::vector<float>* endX = new std::vector<float>;
-    std::vector<float>* endY = new std::vector<float>;
-    std::vector<float>* endZ = new std::vector<float>;
-
-    inputTree->SetBranchAddress("v_eta", &v_eta);
-    inputTree->SetBranchAddress("v_phi", &v_phi);
-    inputTree->SetBranchAddress("mat_step_length", &stepLength);
-
-    inputTree->SetBranchAddress("mat_X0", &stepX0);
-    inputTree->SetBranchAddress("mat_L0", &stepL0);
-    inputTree->SetBranchAddress("mat_A", &stepA);
-
-    inputTree->SetBranchAddress("mat_sx", &startX);
-    inputTree->SetBranchAddress("mat_sy", &startY);
-    inputTree->SetBranchAddress("mat_sz", &startZ);
-
-    inputTree->SetBranchAddress("mat_ex", &endX);
-    inputTree->SetBranchAddress("mat_ey", &endY);
-    inputTree->SetBranchAddress("mat_ez", &endZ);
-
-    // Loop over all entries ---------------
-    unsigned int entries = inputTree->GetEntries();
+  if (inputTree == nullptr) {
+    return;
+  }
+
+  // Get the different atomic numbers
+  TCanvas* materialCanvas =
+      new TCanvas("materialCanvas", "Materials", 100, 100, 620, 400);
+  materialCanvas->cd();
+  // Draw all the atomic elements & get the histogram
+  inputTree->Draw("mat_A>>hA(100,0.5,100.5)");
+  TH1F* histA = dynamic_cast<TH1F*>(gDirectory->Get("hA"));
+  histA->Draw();
+
+  auto outputFile = TFile::Open(outFile.c_str(), "recreate");
+
+  float v_eta = 0;
+  float v_phi = 0;
+  std::vector<float>* stepLength = new std::vector<float>;
```

Comment:
Are those vectors ever properly destructed? However, since this is not really part of the PR, maybe not that important...

Reply:
I fear not and there seem to be a few places in this code were we do not care about deallocation :skull: I would leave it as it is for now